dotnet/src/Microsoft.Agents.AI.Hosting.OpenAI/EndpointRouteBuilderExtensions.ChatCompletions.cs

```diff
@@ -20,47 +18,46 @@ public static partial class MicrosoftAgentAIHostingOpenAIEndpointRouteBuilderExt
+    /// <param name="agentName">The name of <see cref="AIAgent"/> to map the OpenAI ChatCompletions endpoints for.</param>
     /// <param name="path">Custom route path for the chat completions endpoint.</param>
-    public static void MapOpenAIChatCompletions(
-        this IEndpointRouteBuilder endpoints,
-        string agentName,
-        [StringSyntax("Route")] string? path = null)
+    public static IEndpointConventionBuilder MapOpenAIChatCompletions(this IEndpointRouteBuilder endpoints, string agentName, string? path = null)
     {
-        ArgumentNullException.ThrowIfNull(endpoints);
-        ArgumentNullException.ThrowIfNull(agentName);
-        if (path is null)
-        {
-            ValidateAgentName(agentName);
-        }
-
         var agent = endpoints.ServiceProvider.GetRequiredKeyedService<AIAgent>(agentName);
-
-        path ??= $"/{agentName}/v1/chat/completions";
-        var chatCompletionsRouteGroup = endpoints.MapGroup(path);
-        MapChatCompletions(chatCompletionsRouteGroup, agent);
+        return MapOpenAIChatCompletions(endpoints, agent, path);
     }
 
-    private static void MapChatCompletions(IEndpointRouteBuilder routeGroup, AIAgent agent)
-    {
-        var endpointAgentName = agent.DisplayName;
-        var chatCompletionsProcessor = new AIAgentChatCompletionsProcessor(agent);
+    /// <summary>
+    /// Maps OpenAI ChatCompletions API endpoints to the specified <see cref="IEndpointRouteBuilder"/> for the given <see cref="AIAgent"/>.
+    /// </summary>
+    /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the OpenAI ChatCompletions endpoints to.</param>
+    /// <param name="agent">The <see cref="AIAgent"/> instance to map the OpenAI ChatCompletions endpoints for.</param>
+    public static IEndpointConventionBuilder MapOpenAIChatCompletions(this IEndpointRouteBuilder endpoints, AIAgent agent)
+        => MapOpenAIChatCompletions(endpoints, agent, path: null);
 
-        routeGroup.MapPost("/", async (HttpContext requestContext, CancellationToken cancellationToken) =>
-        {
-            var requestBinary = await BinaryData.FromStreamAsync(requestContext.Request.Body, cancellationToken).ConfigureAwait(false);
+    /// <summary>
+    /// Maps OpenAI ChatCompletions API endpoints to the specified <see cref="IEndpointRouteBuilder"/> for the given <see cref="AIAgent"/>.
+    /// </summary>
+    /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the OpenAI ChatCompletions endpoints to.</param>
+    /// <param name="agent">The <see cref="AIAgent"/> instance to map the OpenAI ChatCompletions endpoints for.</param>
+    /// <param name="path">Custom route path for the chat completions endpoint.</param>
+    public static IEndpointConventionBuilder MapOpenAIChatCompletions(
+        this IEndpointRouteBuilder endpoints,
+        AIAgent agent,
+        [StringSyntax("Route")] string? path = null)
```

Comment:
IMO we should avoid default parameters where possible. I rather have overloads. It makes it easier to modify things later.